src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -0,0 +1,512 @@
+   *     type {@code t}
+   */
+  private static Set<TypedOperation> getProducers(Type t, Type startingType) {
+    Set<Type> processed = new HashSet<>();
+    boolean isSearchingForTargetType = true;
+    List<TypedOperation> result = new ArrayList<>();
+    Set<Type> producerParameterTypes = new HashSet<>();
+    Queue<Type> workList = new ArrayDeque<>();
+    workList.add(startingType);
+
+    // Search for constructors/methods that can produce the specified type.
+    while (!workList.isEmpty()) {
+      Type currentType = workList.poll();
+
+      // Log the unspecified classes that are used in demand-driven input creation.
+      logIfUnspecified(currentType);
+
+      // Only consider the type if it is not a primitive type and if it hasn't already been
+      // processed.
+      if (!processed.contains(currentType) && !currentType.isNonreceiverType()) {
+        Class<?> currentClass = currentType.getRuntimeClass();
+        List<Executable> executableList = new ArrayList<>();
+
+        // Addi constructors if the current type is what we are looking for.
+        if (t.isAssignableFrom(currentType) && !Modifier.isAbstract(currentClass.getModifiers())) {
+          Collections.addAll(executableList, currentClass.getConstructors());
+        }
+
+        // Add methods that return the current type.
+        Collections.addAll(executableList, currentClass.getMethods());
+
+        // The first call checks for methods that return the specified type. Subsequent calls
+        // check for methods that return the current type.
+        Type returnType = isSearchingForTargetType ? t : currentType;
+        for (Executable executable : executableList) {
+          if (executable instanceof Constructor
+              || (executable instanceof Method
+                  && returnType.isAssignableFrom(
+                      Type.forClass(((Method) executable).getReturnType())))) {
+
+            // Obtain the input types and output type of the executable.
+            List<Type> inputTypeList = classArrayToTypeList(executable.getParameterTypes());
+            // If the executable is a non-static method, add the receiver type to
+            // the front of the input type list.
+            if (executable instanceof Method && !Modifier.isStatic(executable.getModifiers())) {
+              inputTypeList.add(0, new NonParameterizedType(currentClass));
+            }
+            TypeTuple inputTypes = new TypeTuple(inputTypeList);
+            CallableOperation callableOperation =
+                executable instanceof Constructor
+                    ? new ConstructorCall((Constructor<?>) executable)
+                    : new MethodCall((Method) executable);
+            NonParameterizedType declaringType = new NonParameterizedType(currentClass);
+            TypedOperation typedClassOperation =
+                new TypedClassOperation(callableOperation, declaringType, inputTypes, returnType);
```

Comment:
Can you abstract out some or all of this?  (At least the code that computes `inputTypes`.)
Could you just call `TypedOperation.forConstructor` or `TypedOperation.forMethod`?  Or, if this is different in some way, please document why you cannot use them.